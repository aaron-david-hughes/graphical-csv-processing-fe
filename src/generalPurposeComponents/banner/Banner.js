import React from 'react';

class Banner extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            id: props.id,
            msg: props.msg,
            color: props.color,
            timeToLive: props.timeToLive
        }
    }

    componentDidMount() {
        setTimeout(() => this.props.removeBanner(this.state.id), this.state.timeToLive);
    }

    render() {
        let backgroundColor;
        let borderColor;

        switch(this.props.type) {
            case 'success':
                borderColor = 'green';
                backgroundColor = '#ccffcc';
                break;
            case 'failure':
                borderColor = 'darkred';
                backgroundColor = '#ffcccc';
                break;
            case 'warning':
                borderColor = 'darkorange';
                backgroundColor = '#ffe3b3';
                break;
            case 'neutral':
            default:
                borderColor = 'black';
                backgroundColor = '#bfbfbf';
        }

        return <div
            style={{
                width: '100%',
                height: '5%',
                backgroundColor: backgroundColor,
                color: borderColor,
                textAlign: 'center',
                border: `1px solid ${borderColor}`
            }}
        >
            <p
                style={{
                    margin: 'unset',
                    padding: '5px'
                }}
            >
                {this.state.msg}
            </p>
        </div>
    }
}

export default Banner;