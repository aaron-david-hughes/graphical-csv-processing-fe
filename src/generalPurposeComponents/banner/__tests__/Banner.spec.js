import React from 'react';
import {shallow} from 'enzyme';
import Banner from '../Banner';

describe('Banner tests', () => {
    let props = {
        type: 'failure',
        id: 'testId',
        msg: 'test message',
        timeToLive: 1000,
        removeBanner: jest.fn()
    };

    let render;
    let instance;

    beforeEach(() => {
       render = shallow(<Banner {...props} />);
       instance = render.instance();
    });

    afterEach(() => {
        jest.useRealTimers();
    });

    describe('Banner render tests', () => {
        it('should match snapshot when failure', () => {
            expect(render).toMatchSnapshot();
        });

        it('should match snapshot when success', () => {
            render = shallow(<Banner {...props} type='success' />);
            expect(render).toMatchSnapshot();
        });

        it('should match snapshot when warning', () => {
            render = shallow(<Banner {...props} type='warning' />);
            expect(render).toMatchSnapshot();
        });

        it('should match snapshot when neutral', () => {
            render = shallow(<Banner {...props} type='neutral' />);
            expect(render).toMatchSnapshot();
        });
    });

    describe('Banner instance tests', () => {
        it('should call the setTimeout function', () => {
            jest.useFakeTimers();
            jest.spyOn(global, 'setTimeout');

            instance.componentDidMount();

            expect(setTimeout).toHaveBeenCalledTimes(1);
            expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1000);
        });

        it('should call removeBanner after 1 second', () => {
            jest.useFakeTimers();

            instance.componentDidMount();

            jest.runAllTimers();

            expect(props.removeBanner).toHaveBeenCalled();
            expect(props.removeBanner).toHaveBeenCalledWith('testId');
        });
    });
});