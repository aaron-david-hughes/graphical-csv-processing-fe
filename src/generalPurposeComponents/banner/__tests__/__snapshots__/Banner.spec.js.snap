// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner tests Banner render tests should match snapshot when failure 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ffcccc",
      "border": "1px solid darkred",
      "color": "darkred",
      "height": "5%",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <p
    style={
      Object {
        "margin": "unset",
        "padding": "5px",
      }
    }
  >
    test message
  </p>
</div>
`;

exports[`Banner tests Banner render tests should match snapshot when neutral 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#bfbfbf",
      "border": "1px solid black",
      "color": "black",
      "height": "5%",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <p
    style={
      Object {
        "margin": "unset",
        "padding": "5px",
      }
    }
  >
    test message
  </p>
</div>
`;

exports[`Banner tests Banner render tests should match snapshot when success 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ccffcc",
      "border": "1px solid green",
      "color": "green",
      "height": "5%",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <p
    style={
      Object {
        "margin": "unset",
        "padding": "5px",
      }
    }
  >
    test message
  </p>
</div>
`;

exports[`Banner tests Banner render tests should match snapshot when warning 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ffe3b3",
      "border": "1px solid darkorange",
      "color": "darkorange",
      "height": "5%",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <p
    style={
      Object {
        "margin": "unset",
        "padding": "5px",
      }
    }
  >
    test message
  </p>
</div>
`;
